src/index.ts

```diff
@@ -14,15 +15,15 @@ import {RoutingControllersOptions} from "./RoutingControllersOptions";
 /**
  * Registers all loaded actions in your express application.
  */
-export function useExpressServer<T>(expressApp: T, options?: RoutingControllersOptions): T {
+export function useExpressServer(expressApp: express.Express, options?: RoutingControllersOptions): express.Express {
```

Comment:
I know. Do you know what is inheritance in OOP? The `express.Express` extends the `express.Application`:

```typescript
interface Express extends Application {
        /**
            * Framework version.
            */
        version: string;

        /**
            * Expose mime.
            */
        mime: string;

        (): Application;

        /**
        * Create an express application.
        */
        createApplication(): Application;

        createServer(): Application;

        application: any;

        request: Request;

        response: Response;
    }
```
So if we are demanding `express.Express`, user won't be able to pass his own `express.Application`. But thanks to inheritance, if we demand `express.Application` as parameter, you can easily pass there `express.Express` from `express()` :wink:

It looks like that @pleerock didn't use `express.Express` specific methods/properties, so it's not necessary to use `express.Express`.